src/wms/src/main/java/org/geoserver/wms/map/LiteFeatureTypeStyle.java

```diff
@@ -0,0 +1,93 @@
+/* (c) 2024 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wms.map;
+
+import java.awt.Composite;
+import java.awt.Graphics2D;
+import java.util.List;
+import org.geotools.api.filter.expression.Expression;
+import org.geotools.api.filter.sort.SortBy;
+import org.geotools.api.style.Rule;
+import org.geotools.data.util.ScreenMap;
+import org.geotools.map.Layer;
+import org.geotools.renderer.crs.ProjectionHandler;
+
+/**
+ * This is a simple class that contains the information needed to render a layer.
+ *
+ * <p>Basically, for a SLD, you create one of these for each of the FeatureTypeStyles inside it.
```

Comment:
I would make it clear this is a copy of GeoTools code that is here to support building queries for rendering in formats that are not using StreamingRenderer.

Reply:
added